metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/ResourceUtils.java

```diff
@@ -17,12 +28,95 @@ public static Set<String> getAllAspectNames(final EntityService entityService, f
     return entityService.getEntityAspectNames(entityName);
   }
 
-  public static void tryIndexRunId(
-      final Urn urn,
-      final @Nullable SystemMetadata systemMetadata,
+  public static void tryIndexRunId(final Urn urn, final @Nullable SystemMetadata systemMetadata,
       final EntitySearchService entitySearchService) {
     if (systemMetadata != null && systemMetadata.hasRunId()) {
       entitySearchService.appendRunId(urn.getEntityType(), urn, systemMetadata.getRunId());
     }
   }
+
+  public static SearchResult validateSearchResult(final SearchResult searchResult,
+      @Nonnull final EntityService entityService) {
+    if (searchResult == null) {
+      return null;
+    }
+    Objects.requireNonNull(entityService, "entityService must not be null");
+
+    SearchResult validatedSearchResult = new SearchResult().setFrom(searchResult.getFrom())
+        .setMetadata(searchResult.getMetadata())
+        .setPageSize(searchResult.getPageSize());
+
+    SearchEntityArray validatedEntities = searchResult.getEntities()
+        .stream()
+        .filter(searchEntity -> entityService.exists(searchEntity.getEntity()))
+        .collect(Collectors.toCollection(SearchEntityArray::new));
+    validatedSearchResult.setEntities(validatedEntities);
+    validatedSearchResult.setNumEntities(validatedEntities.size());
```

Comment:
I don't think we can do this! 

numEntities represents the TOTAL number of entities that matched a search query. This will only take the single page of results. It will cause the total numbers to report incorrectly from what I can tell.

To be safe, just take num entities from the original SearchResult

Reply:
Sounds good!